auditFile/fileHandle.js

```diff
@@ -50,6 +50,18 @@ function readAndValidate (pathOverride) {
   if (!versions[version]) {
     throw Error(`Unrecognized ${FILE.FILENAME} content version ${version}`)
   }
+  if (data.version > 0 && data.requireReason && data.requireReason !== "false") {
```

Comment:
I'd like to avoid adding top-level keys to the schema. 
Current schema includes a "rules" field where requireReason could exist. 
I'd also prefer to do `=== true` here to avoid ambiguity.

Reply:
> _Feels like I should keep the tests, but move this logic out of Djv validation and into custom logic. Waddayathink?_

@naugtur I went ahead in this direction, and here's a first minimal viable version (this PR, plus CLI handling in https://github.com/naugtur/audit-resolve-core/pull/5 ). How does that look to you?
https://user-images.githubusercontent.com/522085/176704042-bd8960b9-b7d4-4869-924a-2caef2cf6ef8.mp4

Reply:
> I liked it as a **djv validation**.

@naugtur I see no problem coming back to this, but I don't see how I would be able to reach the same CLI-user error quality with it (pointing at the precise entry failing validation). A failed Djv validation results in a *global* error that says the file failed to validate, but then I'm toast to say something more precise than "validation failed, fix what's wrong and I can't tell you what's wrong" to the user. Or am I missing something? Can you point me to something to take inspiration from?

> **Implementation outside** is fine too if it maintains [what] I said [here](https://github.com/naugtur/npm-audit-resolver/pull/61/files#r911390272): I think I want a failure to result from attempting to load a file with missing reasons instead of adding more RESOLUTION types. RESOLUTION is something that could one day be standardized to some extent and work beyond this tool. (wishful thinking probably, but I did try) and I want to keep it generic. It's not impossible to convince me otherwise if you have valid reasons.

I see your point and agree that these new RESOLUTIONs are unlike the other ones, and should be separated into another concern. So, if proceeding with this approach (not Djv validation), will move away from resolutions. That should be doable with a bit of extra argument passing and an extra getter.

> As for **implementing rules as env** - We need a separate PR for this to design what they'd look like - I'd hope for them to be somewhat consistent with the rules field in the file. How about an env var that lets you merge/override the values in rules? We'd keep the names as an yet unspecified list and letter case consistency (since override would be provided as a json string)

A json string override seems legit. However, I'm not sure what you mean by _"We'd keep the names as an yet unspecified list and letter case consistency"_ , can you precise / reword?

Also, should we have:

- A merge (`{a=1, b=2, c=3} + {a=10, b=10} = {a=10, b=10, c=3}`) ?
- Or a full override (`{a=1, b=2, c=3} + {a=10, b=10} = {a=10, b=10}`) ?

Reply:
Yeah, I was concerned djv would not give you enough details. I think if you dig deep in that error you get it has some metadata about where in the JSON the error happened, but it won't be enough to say why exactly it failed. I did use custom validations in a different project and got the error message to "barely enough" level. 
Since we're agreed on not adding new RESOLUTIONs I'm now happy with both versions djv or not.

I'd go with merge not override.

Reply:
> Since we're agreed on not adding new RESOLUTIONs I'm now happy with both versions djv or not.

@naugtur implemented and pushed. How does that look?

> I'd go with merge not override.

👍 but not implemented yet